c++/test/TestTimezone.cc

```diff
@@ -420,9 +421,21 @@ namespace orc {
 #endif
   }
 
+  char* deepcopy(const char* name) {
+    // this allocates a new buffer that must be freed after use
+#ifdef _MSC_VER
+    return _strdup(name);
+#else
+    return strdup(name);
+#endif
+  }
+
   TEST(TestTimezone, testMissingTZDB) {
-    const char* tzDirBackup = std::getenv("TZDIR");
-    if (tzDirBackup != nullptr) {
+    const char* tzDir = std::getenv("TZDIR");
+    char* tzDirBackup = nullptr;
+    if (tzDir != nullptr) {
+      // avoid that unsetting environment variable wrecks pointer to tzDir
+      tzDirBackup = deepcopy(tzDir);
```

Comment:
Could we simply use `std::string tzDirBackup = tzDir;` In this way, we don't need to free it in the end.

Reply:
Done.

Reply:
Sorry for the newbie mistakes; I had checked that `std::string` can be initialized with `nullptr` (even though I now get that it's against the design), but not that it cannot be compared. 😅 

Anyway, using `s.empty()` [works](https://en.cppreference.com/w/cpp/string/basic_string/empty) even without an explicit initialization of `std::string`.